### PR TITLE
Updates to Confluence module for verbatim text and page anchor links

### PR DESCRIPTION
**Pull request by kavian72**

### CODE REVIEW

**Comment by jmcnamara** [comment]
Thanks for this. Note, there are some failing tests that should be fixed before I can merge this. I'll also make some comments inline.

File: lib/Pod/Simple/Wiki/Confluence.pm [hunk 1]
```diff
@@ -174,7 +174,10 @@ sub _end_L {
 
     # Handle links that are parsed as Pod links.
     if ( defined $link_section ) {
-        $link_target = $link_section;
+        # Other than for L<Link Text>, having a link section defined means
```
Comment (jmcnamara) [inline lib/Pod/Simple/Wiki/Confluence.pm]: This comment is a little confusing. Maybe reword it something like:

```perl
# Look for page anchors in the link.
```

File: lib/Pod/Simple/Wiki/Confluence.pm [hunk 2]
```diff
@@ -174,7 +174,10 @@ sub _end_L {
...
+        # that this is a page anchor link.
+        $link_target = ( $link_attrs->{raw} eq "$link_section" ) ? '' : '#';
```
Comment (jmcnamara) [inline lib/Pod/Simple/Wiki/Confluence.pm]: In general I avoid complex ternary assignments for maintenance reasons. Simpler ones are okay. In this case I would use something like:

```perl
if ( defined $link_attrs->{raw} and $link_attrs->{raw} ne $link_section ) {
    $link_target = '#';
}
```
Also, this is the source of the [failing tests](https://travis-ci.org/jmcnamara/pod-simple-wiki/builds/170000985).

File: t/confluence_links.t [hunk 3]
```diff
@@ -13,7 +13,7 @@
 use strict;
 
 use Pod::Simple::Wiki;
-use Test::More tests => 4;
+use Test::More tests => 8;
```
Comment (jmcnamara) [inline t/confluence_links.t]: Thanks for the additional tests. ++

File: lib/Pod/Simple/Wiki/Confluence.pm [hunk 4]
```diff
@@ -103,7 +103,9 @@ sub _handle_text {
     my @tokens = split /(\s+)/, $text;
 
     # Escape any tokens here, if necessary.
-    @tokens = map { s/([[{*_-])/\\$1/g; $_ } @tokens;
+    unless( $self->{_in_Verbatim} ) {
```
Comment (jmcnamara) [inline lib/Pod/Simple/Wiki/Confluence.pm]: As a general rule I only use `unless` as a postfix equality. I think I got that from Perl Best Practices. So use this instead:

```perl
if ( ! $self->{_in_Verbatim} ) {
```
Also, this seems to fix issue #20. Could you verify that case as well.

File: lib/Pod/Simple/Wiki/Confluence.pm (hunk 2 above)
Comment (kavian72) [inline lib/Pod/Simple/Wiki/Confluence.pm]: I'm having some trouble with the failing tests...the cause of the failures is that in Perl 5.10, 5.12, 5.14, the `$link_attrs` hash doesn't contain the "raw" entry, and that seems to be the only way to differentiate `L<Doc Link>` from `L<"PageAnchor">` or `L</"PageAnchor">`.

Any suggestions?

Comment (jmcnamara) [inline lib/Pod/Simple/Wiki/Confluence.pm]: You would probably need to specify a minimum version of Pod::Simple to ensure that `raw` is supported.

However, that generally leads to a lot of complaints from people with Perl installations where they can't upgrade core modules.

See if there is some other way that will work with the versions of Pod::Simple that shipped with older perl versions.
